### PR TITLE
OTBR config changes for Thread 1.3

### PR DESCRIPTION
**Pull request by suveshpratapa**
- Specify all proper config flags for Thread 1.2 vs Thread 1.3 (and efr or otherwise)
- Specify proper thread protocol version

### CODE REVIEW

**Comment by suveshpratapa** [comment]
@lmnotran Please add anyone else required for review.

File: script/otbr-setup.bash [hunk 1]
```diff
@@ -89,14 +89,23 @@ elif [ "${REFERENCE_RELEASE_TYPE?}" = "1.3" ]; then
...
+                "OTBR_OPTIONS=\"-DOT_THREAD_VERSION=1.3 -DOTBR_DUA_ROUTING=ON -DOT_DUA=ON -DOT_MLR=ON -DOTBR_DNSSD_DISCOVERY_PROXY=ON -DOTBR_SRP_ADVERTISING_PROXY=ON -DOT_SRP_CLIENT=ON -DOT_TREL=ON -DOT_RCP_RESTORATION_MAX_COUNT=100 -DOTBR_PACKAGE_VERSION=${OT_BR_POSIX_COMMIT_HASH} -DOT_PACKAGE_VERSION=${OPENTHREAD_COMMIT_HASH}\""
             )
             ;;
         *)
             readonly BUILD_OPTIONS=(
+                'INFRA_IF_NAME=eth0'
                 'RELEASE=1'
-                'NETWORK_MANAGER=0'
                 'REFERENCE_DEVICE=1'
+                'BACKBONE_ROUTER=1'
```
Comment (lmnotran) [inline script/otbr-setup.bash:100]: Are you sure these options should be set for all vendors?

Comment (suveshpratapa) [reply script/otbr-setup.bash:100]: I am.  However, I think we should have @edmont (or others if you know) review as well.

Comment (simonlingoogle) [reply script/otbr-setup.bash:100]: Should be mandatory to have BACKBONE_ROUTER for 1.3 BR to support Thread 1.2 Multicast Forwarding.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Comment by suveshpratapa** [comment]
@lmnotran Any other comments, can we merge this PR?

**Comment by suveshpratapa** [comment]
@simonlingoogle @edmont Can one of you please review this change and get it in? We need it for 1.3 OTBR cert tests. (@lmnotran is on vacation)

**Review by lmnotran — APPROVED** [review]

**Comment by suveshpratapa** [comment]
@jwhui @simonlingoogle Can we merge this change? It is required to produce proper Raspi images for 1.3 certification.

Additionally, there's an open THCI PR by @JaneFromSilabs ([7806](https://github.com/openthread/openthread/pull/7806)) that is depending on this change being merged.

File: script/otbr-setup.bash [hunk 2]
```diff
@@ -89,14 +89,23 @@ elif [ "${REFERENCE_RELEASE_TYPE?}" = "1.3" ]; then
                 'DHCPV6_PD=0'
                 'WEB_GUI=0'
                 'REST_API=0'
-                "OTBR_OPTIONS=\"-DOT_BORDER_ROUTING=ON -DOT_FULL_LOGS=ON -DOTBR_DUA_ROUTING=ON -DOT_DUA=ON -DOT_MLR=ON -DOTBR_DNSSD_DISCOVERY_PROXY=ON -DOTBR_SRP_ADVERTISING_PROXY=ON -DOT_SRP_CLIENT=ON -DOT_TREL=ON -DOT_RCP_RESTORATION_MAX_COUNT=100 -DOTBR_PACKAGE_VERSION=${OT_BR_POSIX_COMMIT_HASH} -DOT_PACKAGE_VERSION=${OPENTHREAD_COMMIT_HASH}\""
```
Comment (edmont) [inline script/otbr-setup.bash:92]: Having BR log output have been shown to be useful, do we want to remove them?

Comment (suveshpratapa) [reply script/otbr-setup.bash:92]: Thanks! I think the option was removed by accident. Re-added it.

**Review by edmont — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**